### PR TITLE
Resolves #28 - Create IsEmptyEnumerable Matcher

### LINKED ISSUE #28: Create IsEmptyIterable Matcher

### PR DESCRIPTION
**Pull request by jpopadak**

### CODE REVIEW

File: src/CoreMatchers/Descriptions/Description.cs [hunk 1]
```diff
@@ -19,6 +19,16 @@ public IDescription AppendList(string before, string separator, string after, pa
...
         }
 
+        public IDescription AppendList(string before, string separator, string after, IEnumerable<IDescribable> args)
+        {
+            return appendList(before, separator, after, args);
+        }
+
+        public IDescription AppendValueList<T>(string start, string separator, string end, IEnumerable<T> values)
+        {
+            return AppendList(start, separator, end, new SelfDescribingValueEnumerable<T>(values));
```
Comment (TimeDelta) [inline src/CoreMatchers/Descriptions/Description.cs]: add a conditional check to ensure that `T` cannot already be treated as a `SelfDescribingValue`. If that is the case then we don't need to hit the existing constructor that wraps all the values in `SelfDescribingValue`. Create a new constructor for `SelfDescribingValueEnumerable` that accepts something like `IEnumerable<T> where T : SelfDescribingValue`. Disclaimer: I'm not sure about the syntax.

File: src/CoreMatchers/Descriptions/SelfDescribingValueEnumerable.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            }
+            this.values = describingValues;
+        }
+
+        public IEnumerator<IDescribable> GetEnumerator()
+        {
+            return values.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
```
Comment (TimeDelta) [inline src/CoreMatchers/Descriptions/SelfDescribingValueEnumerable.cs]: what's with the "." in the middle of this method name? and where is this used?

Comment (jpopadak) [inline src/CoreMatchers/Descriptions/SelfDescribingValueEnumerable.cs]: This is called implementing an interface explicitly. It is required by some interfaces.
http://stackoverflow.com/a/143423/6079304

Comment (TimeDelta) [inline src/CoreMatchers/Descriptions/SelfDescribingValueEnumerable.cs]: ok

**Comment by jpopadak** [comment]
@jpopadak We need to ensure we put these in the correct subfolder for the StandardMatcher project

**Pull request closed without merging**